yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtilSuite.scala

```diff
@@ -233,4 +234,12 @@ class YarnSparkHadoopUtilSuite extends SparkFunSuite with Matchers with Logging
       }
     assert(caught.getMessage === "Can't get Master Kerberos principal for use as renewer")
   }
+
+  test("check different hadoop utils based on env variable") {
+    System.setProperty("SPARK_YARN_MODE", "true")
+    assert(SparkHadoopUtil.get.getClass.getSimpleName === "YarnSparkHadoopUtil")
```

Comment:
While we're here, comparing `...getClass === classOf[...]` would be better IMO. Fewer magic strings.